### ISSUE TITLE
Fix last names that are also titles, e.g. Dr. Martin Luther King Jr.

### ISSUE THREAD

**Issue by gabekneisley** [derek73/python-nameparser#27]
The title is set to 'Dr. King', as if Dr. King were literally a king.
The last name is set to 'Jr.'
The suffix is empty.

`nameparser==0.3.3` in requirements
running inside of Django 1.7 in a virtualenv

**Comment by derek73** [comment]
The parser allows titles to be chained to recognize things like "vice president" and "secretary general". Because "dr" and "king" are both titles, the parser counts them both as titles. If we only give the parser one name we assume it's a last name, so because there's only on part that isn't a title, Jr. is treated as a last name. 

I think this is a limitation of a simple rule based approach. I tried messing with it but couldn't think of anything that wouldn't break other tests.

You can fix this in your code by removing "king" from the titles constant if "king" is not a common title in your dataset.

```
    >>> from nameparser.config import CONSTANTS
    >>> CONSTANTS.titles.remove('king')
```

**Comment by gabekneisley** [comment]
Thanks Derek, that was precisely my workaround. Just to be clear, my code looked like this:

```python
from nameparser import HumanName
mlk = HumanName('Dr. Martin Luther King Jr.')
mlk.title # u'Dr. King'
mlk.last # u'Jr.'
```

This would mean anyone with a last name that is also a title (King, Duke, Sargent, Queen, Prince) would be misidentified. Perhaps this could be fixed with a list of titles that collide with last names, and thus must be the first token or preceded with a comma to registered as a title. This way, King Richard III or Richard III, King of England would be: `{'title': 'King', 'first': 'Richard', 'suffix': 'III'}`, while Richard Allen King Sr. would be `{'first': 'Richard', 'middle': 'Allen', 'last': 'King': 'suffix': 'Sr.'}` rather than `{title: 'King', 'first': 'Richard', 'last': 'Allen': 'suffix': 'Sr.'}`

Either way, thanks for the library. It is awesome, and it is fantastic that it took me about 10 minutes of reading the docs to fix my issue.

**Comment by derek73** [comment]
There's a similar workaround for titles that are first name titles so maybe it does make sense to have a list of collisions with last names. There's so many titles though, almost any noun could be a title in some situations. 

"jr." as the last name seems a tempting place to look for a more agnostic way to fix it. That's already a known suffix, so seems like if I could just prevent that from going in the last name bucket and then make sure that last gets a higher priority than title, the name would parse correctly. That's what I tried to do but it didn't work out. Maybe it will come to me in my sleep someday.

**Comment by derek73** [comment]
Fixed in v0.3.9
